libraries/AP_GPS/AP_GPS_SBP2.h

```diff
@@ -0,0 +1,184 @@
+        uint32_t tow;         //< GPS Time of Week (unit: ms)
+        double lat;           //< Latitude (unit: degrees)
+        double lon;           //< Longitude (unit: degrees)
+        double height;        //< Height (unit: meters)
+        uint16_t h_accuracy;  //< Horizontal position accuracy estimate (unit: mm)
+        uint16_t v_accuracy;  //< Vertical position accuracy estimate (unit: mm)
+        uint8_t n_sats;       //< Number of satellites used in solution
+        struct PACKED flags {
+            uint8_t fix_mode:3;  //< Fix mode (0: invalid, 1: SPP, 2: DGNSS, 3: Float RTX, 4: Fixed RTX
+            uint8_t res:4;       //< Reserved
+            bool raim_repair:1;  //< Solution from RAIM?
+        } flags;
+    }; // 34 bytes
+
+    // Velocity in NED Velocity in local North East Down (NED) coordinates.
+    struct PACKED sbp_vel_ned_t {
+        uint32_t tow;          //< GPS Time of Week (unit: ms)
+        int32_t n;             //< Velocity North coordinate (unit: mm/s)
+        int32_t e;             //< Velocity East coordinate  (unit: mm/s)
+        int32_t d;             //< Velocity Down coordinate  (unit: mm/s)
+        uint16_t h_accuracy;   //< Horizontal velocity accuracy estimate (unit: mm/s)
+        uint16_t v_accuracy;   //< Vertical velocity accuracy estimate (unit: mm/s)
+        uint8_t n_sats;        //< Number of satellites used in solution
+        struct PACKED flags {
+            uint8_t fix_mode:3;  //< Fix mode (0: invalid, 1: SPP, 2: DGNSS, 3: Float RTX, 4: Fixed RTX
+            uint8_t res:5;       //< Reserved
+        } flags;
+    }; // 22 bytes
+
+    void _sbp_process();
+    void _sbp_process_message();
+    bool _attempt_state_update();
+
+    // ************************************************************************
+    // Internal Received Messages State
+    // ************************************************************************
+    uint32_t last_heartbeat_received_ms;
+    uint32_t last_injected_data_ms;
+
+    struct sbp_heartbeat_t last_heartbeat;
+    struct sbp_gps_time_t  last_gps_time;
+    struct sbp_dops_t      last_dops;
+    struct sbp_pos_llh_t   last_pos_llh;
+    struct sbp_vel_ned_t   last_vel_ned;
+
+    uint32_t               last_full_update_tow;
+    uint16_t               last_full_update_wn;
+
+    // ************************************************************************
+    // Monitoring and Performance Counting
+    // ************************************************************************
+
+    uint32_t crc_error_counter;
+
+    // ************************************************************************
+    // Logging to DataFlash
+    // ************************************************************************
+
+    void logging_log_full_update();
+    void logging_log_raw_sbp(uint16_t msg_type, uint16_t sender_id, uint8_t msg_len, uint8_t *msg_buff);
+
+
+};
+
+int32_t distMod(int32_t tow1_ms, int32_t tow2_ms, int32_t mod);
```

Comment:
I would be fine with it being static, but it would need to be part of the class, it wouldn't make much sense to have it as a global function.